### ISSUE TITLE
bridge plugin, Cc field with list email address, repetitively sends emails

### ISSUE THREAD

**Issue by bretmaverick999** [justingit/dada-mail#698]
The contents of a single email can cause Dada/Bridge to get into a mode where it repetitively sends emails.

My club's website is running Dada version 7.2.1. When I last upgraded that was the latest version. We have 10 email lists that use the bridge plugin. Only club members can subscribe to the email lists, and only a subscriber to a list can send emails to the list. This is the second time one of the lists got stuck repetitively sending emails. The first time it happened several years ago I was traveling and via smart phone got the list shut down but couldn't debug what happened. 

It recently happened again, and this time I had email logging enabled. Working with the person who sent the email that caused the problem, and a "test" list, we were able to recreate the problem. I then spent significant time narrowing down the conditions that cause the problem. 

To cause the list to continuously send emails the list subscriber creates an email and fills in the "Cc:" field with the email address for the list (such as "list1@b.com"). Most other stuff doesn't matter. When I try this scenario from multiple personal email accounts the problem doesn't happen. It only happens from his account. To simplify recreating the problem I had the person send an email to the email address for my test list, and before bridge could process the email I saved a copy of the file. When bridge ran it recreated the problem, so I no longer needed him to send emails for me as I had the file with the saved copy of the email in it. Normally when bridge runs it sends the email to all of the list subscribers and then removes the email from the file (typically leaving an file empty). When the problem happens though the email is sent to each list subscriber via the To field, but the Cc field has the email address for the list, thus causing the email to be added to the end of the file for the list.

If I copy the following lines to the file for the email list, when bridge runs, it will see that the sender of the email is a subscriber to the list, and then send the email to each of the subscribers
```
From a.a@gmail.com  Tue Oct 17 12:02:36 2017
Return-Path: <a.a@gmail.com>
Date: Tue, 17 Oct 2017 12:02:33 -0600
From: a.a@gmail.com
Cc: wmc-test@wasatchmountainclub.org
Subject: minimal email that fails

this is a test

```
(Note - one blank line follows the text of the email)
After bridge runs, instead of the file containing 0 emails it contains 3 emails to be sent (1 for the owner of the list, and my test list has 2 subscribers so one for each of them). I've configured Bridge to only process 1 email per execution. After Bridge runs again it leaves 2 emails in the file, and then 1 email in the file, and then 0.  Yes eventually Bridge will drain the file of emails. But if the list has 300 subscribers then each subscriber will receive 301 copies of the email. Since my cron runs Bridge every 2 minutes that's 10 hours of emails, and subscribers start panicking and call me after 30 or so.

I looked thru all the open bugs and closed ones back thru roughly version 5.x but couldn't find any like this one.

**Comment by justingit** [comment]
I would def. upgrade to the latest (v10.7.1) as a first step - there's a lot of ways an infinite loop can happen, and I do remember this being a case we squashed.

**Comment by bretmaverick999** [comment]
After creating this issue I had problems with 7.2.1 so I needed to upgrade. I tired to upgrade to 10.7.1 but found out 8.x.x onward only supported the database backend storage method. My 7.2.1 version used text files. To upgrade from the text files to the database backend was to much work to bite off at that time. Hence I upgraded to 7.4.1. And that fixed the problems I was having. 

About 3 weeks ago the problem I documented above happened again. Here's how I have things configured. 
- I'm using Bridge,
- Discussion List is enabled, 
- p.p. mode is checked with the From: header line configured with:
< !-- tmpl_var list_settings.list_name -- > for '< !-- tmpl_var original_from_phrase default="Subscriber" -- >'
- Replies to messages should = be addressed to the sender

In looking thru the ~/.dada_files/.tmp/bridge_received_msgs-test.mbox I can see what happened, and it's not a run away list. Instead, just like the issue described in https://github.com/justingit/dada-mail/issues/507 , _it seems like Dada isn't recognizing that the email address in the Cc: field is the list email address_!. 

Lets assuming "b.com" is my domain
With 7.4.1 what **NORMALLY** happens when this email
    To: a@a.com
     Cc: test-list@b.com
gets sent to the list, first Dada processes headers and then sends out emails to the members of the list, and the email that the list member receives looks like this:
    To: test-list@b.com
     Cc: test-list@b.com
Note 1 - **Only one email gets sent to each member of the list** even though the list email address is in the To: and Cc: fields.
Note 2 - If a member of the list clicks Reply the "To" is to "a@a.com", or clicks Reply-All the "To" field is test-list@b.com so everything is working fine.

For some unknown reason, if an **UNUSUAL** email (I documented it above)
    To: a@a.com
     Cc: test-list@b.com
gets sent to the list, first Dada processes the headers and then sends out emails to the members of the list, and the email that the list member receives looks like this:
    To: listmember1@a.com
     Cc: test-list@b.com
**The problem is that Dada sends out two (2!) emails, one to the "To" and one to the "Cc"!**  This is the ROOT problem.
So every email sent by Dada goes to the "To" and "Cc" emails, and thus causes one to come back into the list email address, and then Dada processes it. So if the email list contains 10 email addresses, there's the first UNUSUAL email followed by 10 more copies of it, for a total of 11 emails that are very similar. I am positive this is what's happening as I've recreated it many time using the UNUSUAL email (_which doesn't appear unusual but somehow is_).

Note - I applied the fix from https://github.com/justingit/dada-mail/issues/507, did the UNUSUAL email, and things worked like NORMAL which is understandable as the "Cc" field was not in any emails sent out. **HOWEVER**, for those who receive the email from Dada, if they click Reply or ReplyAll the list email address isn't there so they can't reply back to the list, and instead must manually enter the list email address. 

Now that I'm not horrible swamped I plan on upgrading to the latest code but it will take a while as I have to hook my website into database backend (after our members login to our club website they display a webpage with checkboxes per list, add or remove them from the lists, and click save which currently adds or removes their email from the files for the lists). I have full faith the problem will be fixed, just worried about the Bridge> Discussion List> "Replies to: Sender or List" setting...
